src/NHibernate/Util/EnumerableExtensions.cs

```diff
@@ -6,6 +6,16 @@ namespace NHibernate.Util
 {
 	public static class EnumerableExtensions
 	{
+		public static HashSet<T> ToHashSet<T>(this IEnumerable<T> collection)
```

Comment:
Personally I don't see any problem in having utility extension methods. But I will update my pull request if this is NHibernate's guidelines.